### PR TITLE
Gsea plot

### PR DESCRIPTION
**Pull request by robinpaul85**
## Description

In GSEA app on clicking a row in geneset table, the gsea plot corresponding to that geneset is displayed.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/routes/genesetEnrichment.ts [hunk 1]
```diff
@@ -26,7 +26,15 @@ function init({ genomes }) {
 	return async (req: any, res: any): Promise<void> => {
 		try {
 			const results = await run_genesetEnrichment_analysis(req.query as genesetEnrichmentRequest, genomes)
-			res.send(results as genesetEnrichmentResponse)
+			if (!req.query.geneset_name) {
```
Comment (xzhou82) [inline server/routes/genesetEnrichment.ts:29]: please add some comment at the top to explain the logic. when does it run gsea, when does it render a graph

Comment (robinpaul85) [reply server/routes/genesetEnrichment.ts:29]: I added more comments

File: server/utils/gsea.py [hunk 2]
```diff
@@ -58,16 +84,23 @@ def extract_symbols(x):
...
-            start_gsea_time = time.time()
-            if __name__ == "__main__":
-               result = blitz.gsea(signature, msigdb_library).T
-               print ("result:",result.to_json())
-            stop_gsea_time = time.time()   
-            gsea_time = stop_gsea_time - start_gsea_time
-            print (f"GSEA time: {gsea_time} seconds")
+            try: # Extract ES data to be plotted on client side
+               geneset_name=json_object['geneset_name'] # Checks if geneset_name is present, if yes it indicates the server request is for generating the image. It retrieves the result.pkl file and generates the image without having to recompute gsea again.
+               result = pd.read_pickle(os.path.join(cachedir,"result.pkl"))
```
Comment (xzhou82) [inline server/utils/gsea.py:89]: won't work. multiple requests will corrupt and break it

Comment (robinpaul85) [reply server/utils/gsea.py]: Support for random file names for both the pickle file and the image file have been added.

**Review by xzhou82 — APPROVED** [review]
works thanks

**Pull request merged**